tools/testing/e2e/test_utils.h

```diff
@@ -0,0 +1,154 @@
+#ifndef IREE_TESTING_COMMON_UTILS_H_
+#define IREE_TESTING_COMMON_UTILS_H_
+
+#include <stdio.h>
+
+#include "iree/base/api.h"
+#include "iree/base/internal/flags.h"
+#include "iree/hal/api.h"
+#include "iree/vm/api.h"
+
+IREE_FLAG(bool, require_exact_results, true,
+          "Requires floating point result elements to match exactly.");
+IREE_FLAG(
+    float, acceptable_fp_delta, 1e-5f,
+    "Maximum absolute difference allowed with inexact floating point results.");
+IREE_FLAG(
+    int32_t, max_elements_to_check, 10000,
+    "Maximum number of tensor elements to check for the given test. For larger "
+    "matrices, only every n-th element will be checked for some n chosed to "
+    "stay just under that threshold and to avoid being a divisor of the inner "
+    "dimension size to avoid special patterns. As the check uses a slow "
+    "reference implementation, this is a trade-off between test latency and "
+    "coverage. The value 0 means check all elements.");
+
+const char* emoji(bool good);
```

Comment:
lol, could use more descriptive function and variable names here

Reply:
This was a part of the iree-e2e-matmul-test implementation. It's used in the context here: https://github.com/openxla/iree/blob/6ef1cfe4fe47dadacca62b5bfc96ac5a3a9c4dbf/tools/iree-e2e-matmul-test.cc#L662 It's currently not used by `iree-e2e-conv2d-test`.